File: dostavka_app/lib/models/courier.dart
class Courier {
  final int id;
  final String name;
  final String phone;
  final String status;
  final String createdAt;

  const Courier({
    required this.id,
    required this.name,
    required this.phone,
    required this.status,
    required this.createdAt,
  });

  factory Courier.fromJson(Map<String, dynamic> json) {
    return Courier(
      id: json['id'] as int,
      name: json['name'] as String? ?? '',
      phone: json['phone'] as String? ?? '',
      status: json['status'] as String? ?? 'active',
      createdAt: json['created_at'] as String? ?? '',
    );
  }

  bool get isBlocked => status == 'blocked';
  bool get isActive => status == 'active';

  String get statusLabel => isBlocked ? 'Заблокирован' : 'Активен';
}

File: frontend/src/types/courier.ts
export type CourierStatus = 'active' | 'blocked'

export interface Courier {
  id: number
  name: string
  phone: string
  status: CourierStatus
  created_at: string
}

export interface CourierCreate {
  name: string
  phone: string
}

export const courierStatusConfig: Record<CourierStatus, { label: string; color: string }> = {
  active: { label: 'Активен', color: 'bg-green-100 text-green-800 dark:bg-green-900 dark:text-green-300' },
  blocked: { label: 'Заблокирован', color: 'bg-red-100 text-red-800 dark:bg-red-900 dark:text-red-300' },
}